packages/editor/src/components/post-actions/index.js

```diff
@@ -20,38 +20,57 @@ import { usePostActions } from './actions';
 
 const { Menu, kebabCase } = unlock( componentsPrivateApis );
 
-export default function PostActions( { postType, postId, onActionPerformed } ) {
-	const [ activeModalAction, setActiveModalAction ] = useState( null );
-	const { item, permissions } = useSelect(
+function useEditedEntityRecordsWithPermissions( postType, postIds ) {
+	const { items, permissions } = useSelect(
 		( select ) => {
 			const { getEditedEntityRecord, getEntityRecordPermissions } =
 				unlock( select( coreStore ) );
 			return {
-				item: getEditedEntityRecord( 'postType', postType, postId ),
-				permissions: getEntityRecordPermissions(
-					'postType',
-					postType,
-					postId
+				items: postIds.map( ( postId ) =>
+					getEditedEntityRecord( 'postType', postType, postId )
+				),
+				permissions: postIds.map( ( postId ) =>
+					getEntityRecordPermissions( 'postType', postType, postId )
```

Comment:
This change has a couple of issues:

* Iterating over values here results in different `items` and `permissions` arrays on every `mapSelect` call.
* Technically, it can also result in multiple HTTP calls if entity data isn't available in the store.

You can fetch permissions for multiple entities using  `getEntityRecordsPermissions`, but we don't have a similar substitute for `getEditedEntityRecord`. The `getEntityRecords` could work but will trigger a REST API request for each new post selection in DataViews.

## Screenshot
![CleanShot 2024-12-10 at 20 06 06](https://github.com/user-attachments/assets/4ea73568-1a97-4285-88b4-c5a07873ce71)

Reply:
Thanks @Mamaduka, will push up a fix for this today.

Reply:
Followed up in this PR: https://github.com/WordPress/gutenberg/pull/67872

Reply:
Thanks, @louwie17; I'll have a look later today or tomorrow.